tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -285,14 +285,14 @@ static PyObject * THPVariable_index_scalar(PyObject* self, PyObject* args) {
 static Tensor dispatch_invert(const Tensor & self) {
   AutoNoGIL no_gil;
   OptionalDeviceGuard device_guard(device_of(self));
-  return 1 - self;
+  return self.bitwise_not();
 }
 
 static PyObject * THPVariable_invert(PyObject* self, PyObject* args) {
   HANDLE_TH_ERRORS
   auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
-  if (self_.scalar_type() != at::kByte) {
-    throw TypeError("~ (operator.invert) is only implemented on byte tensors");
+  if (!isIntegralType(self_.scalar_type()) && self_.scalar_type() != at::kBool) {
```

Comment:
Why isn't bool an integral type? It's not different than `uint1`

Reply:
It's defined [here](https://github.com/pytorch/pytorch/blob/cca923c4811dbc00acfe28ac67643a90b723236c/c10/core/ScalarType.h#L214). It's probably worth another independent PR for discussing it...

Reply:
It looks like current usage and definition of `isIntegralType` are more of semantic sense (bool is technically integral, but semantically it is not used as an integer). I would lean toward keep it as it is now.

Examples:

- https://github.com/pytorch/pytorch/blob/cca923c4811dbc00acfe28ac67643a90b723236c/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L43
- https://github.com/pytorch/pytorch/blob/cca923c4811dbc00acfe28ac67643a90b723236c/torch/csrc/autograd/python_variable_indexing.cpp#L171